images/base-notebook/start-notebook

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+import os
+import shlex
+import sys
+
+# If we are in a JupyterHub, we pass on to `start-singleuser` instead so it does the right thing
+if "JUPYTERHUB_API_TOKEN" not in os.environ:
+    print(
```

Comment:
You can change this here to `logging.warning`, or in a separate PR (as you wish).

Reply:
@mathbunnyru yeah, I think that can be handled separately so this becomes more of a 'straight port'.